fastqpreprocessing/src/utilities.cpp

```diff
@@ -0,0 +1,308 @@
+/** 
+ *  @file   utilities.cpp 
+ *  @brief  Utility functions for file processing 
+ *  @author Kishori Konwar 
+ *  @date   2020-08-27 
+ ***********************************************/
+
+#include "utilities.h"
+#include <string>
+#include <cstdint>
+#include <experimental/filesystem>
+
+using namespace std;
+namespace fs = std::experimental::filesystem;
+
+/** @copydoc filesize */
+int32_t filesize(const char *filename) {
+    FILE *f = fopen(filename, "rb");  /* open the file in read only */
+
+    int32_t size = 0;
+    if (fseek(f, 0, SEEK_END) ==0 ) /* seek was successful */
+       size = ftell(f);
+    fclose(f);
+    return size;
+}
+
+/** @copydoc getFileSize */
+int32_t getFileSize(const std::string &fileName) {
+    ifstream file(fileName, ifstream::in | ifstream::binary);
+
+    if (!file.is_open()) {
+        return -1;
+    }
+
+    file.seekg(0, ios::end);
+    int32_t fileSize = file.tellg();
+    file.close();
+
+    return fileSize;
+}
+
+/** @copydoc read_white_list */
+WHITE_LIST_DATA *read_white_list(const string &white_list_file) {
+    // TODO: be careful of caps
+    char ATCG[] = {'A', 'C', 'G', 'T', 'N'};
+
+    fstream newfile;
+    WHITE_LIST_DATA *white_list_data = new  WHITE_LIST_DATA;
+
+    // open a file to perform read operation using file object
+    newfile.open(white_list_file, ios::in);
+    int k = 0;
+    if (newfile.is_open()) {   // checking whether the file is open
+       string tp;
+
+       // read data from file object and put it into string.
+       while (getline(newfile, tp)) {
+        //  cout << tp << "\n"; 
+          //insert the barcode into the list
+          white_list_data->barcodes.push_back(tp);
+
+          for (int i=0; i < tp.size(); i++) {
+            for (int j=0; j < 5; j++) {
+              char c = tp[i];
```

Comment:
more descriptive variable names would be helpful (`corrected` rather than `c` for instance)